<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from robot1_simulation.xacro        | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="robot1_simulation" xmlns:xacro="http://ros.org/wiki/xacro">
    <material name="black">
        <color rgba="0 0 0 1"/>
    </material>
    <link name="base_link"/>
    <joint name="body1_joint" type="fixed">
        <parent link="base_link"/>
        <child link="body_link"/>
        <origin rpy="0 0 0" xyz="0 0 0.075"/>
    </joint>
    <link name="body_link">
        <visual>
        <geometry>
            <box size="0.2 0.3 0.1"/>
        </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <material name="white">
            <color rgba="1 1 1 1"/>
        </material>
        </visual>
        <inertial>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <mass value="0.1"/>
        <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
        </inertial>
        <collision>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.048" radius="0.05"/>
        </geometry>
        </collision>
    </link>
    <gazebo reference="body_link">
        <material>Gazebo/white</material>
    </gazebo>
    <joint name="base_to_wheel1" type="continuous">
        <origin rpy="0 -1.57079632679 0" xyz="0.1 0.1 -0.05"/>
        <parent link="body_link"/>
        <child link="wheel1"/>
        <axis xyz="0 0 1"/>
        <!-- <limit lower="-1" upper="1" effort="1000" velocity="0.5"/> -->
    </joint>
    <link name="wheel1">
        <visual>
        <geometry>
            <cylinder length="0.05" radius="0.05"/>
        </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <material name="black"/>
        </visual>
        <inertial>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <mass value="0.1"/>
        <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
        </inertial>
        <collision>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.048" radius="0.05"/>
        </geometry>
        </collision>
    </link>
    <gazebo reference="wheel1">
        <material>Gazebo/black</material>
    </gazebo>
    <joint name="base_to_wheel2" type="continuous">
        <origin rpy="0 -1.57079632679 0" xyz="-0.1 0.1 -0.05"/>
        <parent link="body_link"/>
        <child link="wheel2"/>
        <axis xyz="0 0 1"/>
        <!-- <limit lower="-1" upper="1" effort="1000" velocity="0.5"/> -->
    </joint>
    <link name="wheel2">
        <visual>
        <geometry>
            <cylinder length="0.05" radius="0.05"/>
        </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <material name="black"/>
        </visual>
        <inertial>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <mass value="0.1"/>
        <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
        </inertial>
        <collision>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.048" radius="0.05"/>
        </geometry>
        </collision>
    </link>
    <gazebo reference="wheel2">
        <material>Gazebo/black</material>
    </gazebo>
    <joint name="base_to_wheel3" type="continuous">
        <origin rpy="0 -1.57079632679 0" xyz="0.1 -0.1 -0.05"/>
        <parent link="body_link"/>
        <child link="wheel3"/>
        <axis xyz="0 0 1"/>
        <!-- <limit lower="-1" upper="1" effort="1000" velocity="0.5"/> -->
    </joint>
    <link name="wheel3">
        <visual>
        <geometry>
            <cylinder length="0.05" radius="0.05"/>
        </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <material name="black"/>
        </visual>
        <inertial>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <mass value="0.1"/>
        <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
        </inertial>
        <collision>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.048" radius="0.05"/>
        </geometry>
        </collision>
    </link>
    <gazebo reference="wheel3">
        <material>Gazebo/black</material>
    </gazebo>
    <joint name="base_to_wheel4" type="continuous">
        <origin rpy="0 -1.57079632679 0" xyz="-0.1 -0.1 -0.05"/>
        <parent link="body_link"/>
        <child link="wheel4"/>
        <axis xyz="0 0 1"/>
        <!-- <limit lower="-1" upper="1" effort="1000" velocity="0.5"/> -->
    </joint>
    <link name="wheel4">
        <visual>
        <geometry>
            <cylinder length="0.05" radius="0.05"/>
        </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <material name="black"/>
        </visual>
        <inertial>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <mass value="0.1"/>
        <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
        </inertial>
        <collision>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <cylinder length="0.048" radius="0.05"/>
        </geometry>
        </collision>
    </link>
    <gazebo reference="wheel4">
        <material>Gazebo/black</material>
    </gazebo>
</robot>

